src/dodal/beamlines/example_beamline.py

```diff
@@ -0,0 +1,120 @@
+from collections.abc import Callable
+
+from bluesky.run_engine import call_in_bluesky_event_loop
+from ophyd_async.core import DEFAULT_TIMEOUT
+from ophyd_async.core import Device as OphydV2Device
+
+from dodal.common.beamlines.beamline_utils import (
+    ACTIVE_DEVICES,
+)
+from dodal.utils import AnyDevice
+
+
+class XYZDetector(OphydV2Device):
+    def __init__(self, prefix: str, *args, **kwargs):
+        self.prefix = prefix
+        super().__init__(*args, **kwargs)
+
+    @property
+    def hints(self):
+        raise NotImplementedError
+
+
+class DeviceInitializationConfig:
+    """
+    eager ones all connect at the beginning in blueapi startup. if they fail to connect we get an error
+    endpoint for what devices are failing
+    only matters in connect failure reporting behavior
+    lazy and eager are not about when connect but whether should be connected
+    """
+
+    eager: bool = True
+    set_name: bool = True
+    default_timeout_for_connect: float = DEFAULT_TIMEOUT
+    default_use_mock_at_connection: bool = False
+
+    def __init__(self, **kwargs):
+        self.eager = kwargs.get("eager", False)
+        self.set_name = kwargs.get("set_name", True)
+        self.default_timeout_for_connect = kwargs.get("timeout", 0)
+        self.default_use_mock_at_connection = kwargs.get(
+            "use_mock_at_connection", False
+        )
+
+
+class DeviceInitializationController:
+    device: AnyDevice | None = None
+    config: DeviceInitializationConfig | None = None
+
+    def __init__(self, config: DeviceInitializationConfig) -> None:
```

Comment:
So this needs to take the factory too:
```suggestion
    def __init__(self, config: DeviceInitializationConfig, factory: Callable[[], AnyDevice]) -> None:
        self.factory = factory
```